bundler/lib/bundler/compact_index_client/updater.rb

```diff
@@ -95,7 +95,12 @@ def checksum_for_file(path)
         # because we need to preserve \n line endings on windows when calculating
         # the checksum
         SharedHelpers.filesystem_access(path, :read) do
-          SharedHelpers.digest(:MD5).hexdigest(File.read(path))
+          File.open(path, "rb") do |f|
```

Comment:
What about to move this into shared helper directly and just call it here? There is no need to complicate this class with logic for "streamed" MD5 IMHO.

Reply:
That makes sense to me.

...I wonder if we could use [Gem::Package::DigestIO](https://github.com/rubygems/rubygems/blob/master/lib/rubygems/package/digest_io.rb) here to calculate the checksum as we write the file rather than performing an additional step. It would replace `copy_file` so might cancel out the benefit.

Reply:
@simi @segiddins I got a DigestIO based solution working over at #7039.

Reply:
I think since it's only used here, it's better if we leave it. There's nothing else doing this pattern in bundler that I could find.